src/components/pages/movies-page/index.tsx

```diff
@@ -0,0 +1,174 @@
+
+
+    async function getMoviesApi(searchValue: string = "" ,pageNumber:number) {
+        setLoader(true)
+        try {
+            history.push(`/page/${searchMovieApi.searchMovie}/${pageNumber}`)
+            const moviesUrl = `http://www.omdbapi.com/?s=${searchValue}&apikey=4f7462e2&page=${pageNumber}`
+            const { data } = await axios.get(moviesUrl);
+            if (data.Response === "False") throw Error("No Data From Api")
+            setMovies(data.Search)
+        } catch ({ message }) {
+            setAlertConfig({ show: true, message })
+            //@ts-ignore
+            setMovies([])
+            clearErrorMessage()
+        } finally {
+            setLoader(false)
+        }
+    }
+
+
+    function clearErrorMessage() {
+        setTimeout(() => {
+            setAlertConfig({ show: false, message: "" })
+        }, 2000);
+    }
+    useEffect(() => {
+        //this code will run: cases:
+        // on initial render
+        // on any chnage in the array params
+        getMoviesApi("scream",1)
+        
+    }, [starsColor])
+
+    function clearMovies() {
+        setMovies([])
+    }
+
+    function revert() {
+        const deletedMoviesCopy = [...deletedMovies];
+        if (!deletedMoviesCopy.length) return;
+        const getLastRevertMovie = deletedMoviesCopy[0];
+        deletedMoviesCopy.splice(0, 1)
+        setMovies([...movies, getLastRevertMovie])
+        setDeletedMovies([...deletedMoviesCopy])
+    }
+
+    function addMovie() {
+        setMovies([...movies]) //example to show state - data[0] = from FORM
+    }
+
+
+    function deleteMovie(moovieId: string): void {
+        const moviesCopy = [...movies]
+        const [index, m] = getMoviesData()
+        moviesCopy.splice(index, 1);
+        setMovies(moviesCopy)
+        setDeletedMovies([...deletedMovies, m])
+        function getMoviesData(): Array<any> {
+            const index = movies.findIndex(m => m.imdbID === moovieId);
+            const movie = movies.find(m => m.imdbID === moovieId);
+            return [index, movie]
+
+        }
+    }
+    function filterOperationApi(value: string) {
+        getMoviesApi(value,1)
+        setMoviePage({page:1})
+        setSearchMovieApi({searchMovie: value})
+    }
+
+    function filterOperation(value: string) {
+        if (!value) return setMovies(movies);
+        const filteredMovies = movies.filter(movie => movie.Title.toLowerCase().includes(value))
+        setMovies(filteredMovies)
+    }
+    
+    async function movieNextPage(){
+    
+       //const [moviepage,setMoviePage] =useState({moviedefault:true,page:1})
+       setMoviePage({page:++moviepage.page})
+       getMoviesApi(searchMovieApi.searchMovie,moviepage.page)
+    //    const {data}= await axios.get(`http://www.omdbapi.com/?s=${searchMovieApi.searchMovie}&plot=short&apikey=4f7462e2&page=${moviepage.page}`)
+    //    setMovies(data.Search)
+       history.push(`/page/${searchMovieApi.searchMovie}/${moviepage.page}`)
```

Comment:
you dont need to route to another page
you should request for another page from the server 
and just replace the content result = [Movies from page  2] to result = [Movies from page  2 ]